### PR TITLE
09/30/2020 

### PR DESCRIPTION
**Pull request by rohitrna**
09/23/2020 re added admin page, password change screen, and log in functionality

### CODE REVIEW

File: IAssets/IAssets/PasswordChangeViewController.swift [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+       }
+    
+    
+    
+    @IBAction func passwordSubmissionButton(_ sender: Any) {
+        let password = self.isValidPassword(testStr: passwordTextField.text!)
+        let passwordConfirmation = self.isValidPassword(testStr: passwordConfirmationTextField.text!)
+     
+        
+        if(password == true && passwordConfirmation == true){
```
Comment (rohitrna) [inline IAssets/IAssets/PasswordChangeViewController.swift:49]: one space before opening brace

File: IAssets/IAssets/PasswordChangeViewController.swift [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        
+        if(passwordTextField.text == passwordConfirmationTextField.text){
+            
+            let appDelegate = UIApplication.shared.delegate as! AppDelegate
+            let containerOfAppDelegate = appDelegate.persistentContainer
+            
+             if let userEntity = NSEntityDescription.insertNewObject(forEntityName: "User", into: containerOfAppDelegate.viewContext) as? User {
```
Comment (rohitrna) [inline IAssets/IAssets/PasswordChangeViewController.swift:56]: need to update "user" with new password

File: IAssets/IAssets/PasswordChangeViewController.swift [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+             }
+             let context = containerOfAppDelegate.viewContext
+             do {
+                 try context.save()
+             } catch let error {
+                 print("Could not save \(error.localizedDescription)")
+             }
+            
+            
+            //userEntity.password = passwordTextField.text
```
Comment (rohitrna) [inline IAssets/IAssets/PasswordChangeViewController.swift:69]: remove commented code

File: IAssets/IAssets/PasswordChangeViewController.swift [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+            
+            
+            
+        }else{
+            self.showAlert(title: "Passwords do not match", message: "Password should contain atleast 8")
```
Comment (rohitrna) [inline IAssets/IAssets/PasswordChangeViewController.swift:74]: correct message

File: IAssets/IAssets/PasswordChangeViewController.swift [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+        }
+        }else{
+            self.showAlert(title: "Invalid Password", message: "Please Enter Valid Password")
+        }
+        
+    }
+    
+    /*
```
Comment (rohitrna) [inline IAssets/IAssets/PasswordChangeViewController.swift:82]: remove cemented code

File: IAssets/IAssets/PasswordChangeViewController.swift [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+import UIKit
+import CoreData
+
+class PasswordChangeViewController: UIViewController {
+    var user: User?
+    
+    @IBOutlet weak var passwordTextField: UITextField!
+    
+    @IBOutlet weak var passwordConfirmationTextField: UITextField!
+
```
Comment (aswathyur) [inline IAssets/IAssets/PasswordChangeViewController.swift:18]: Add proper access specifiers private or fileprivate

File: IAssets/IAssets/PasswordChangeViewController.swift [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+    
+    
+    @IBAction func passwordSubmissionButton(_ sender: Any) {
+        let password = self.isValidPassword(testStr: passwordTextField.text!)
+        let passwordConfirmation = self.isValidPassword(testStr: passwordConfirmationTextField.text!)
+     
+        
+        if(password == true && passwordConfirmation == true){
+        
+        if(passwordTextField.text == passwordConfirmationTextField.text){
```
Comment (aswathyur) [inline IAssets/IAssets/PasswordChangeViewController.swift:51]: One space before opening brace

File: IAssets/IAssets/MyAssetsTableViewController.xib [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+        <tableView opaque="NO" clipsSubviews="YES" clearsContextBeforeDrawing="NO" contentMode="scaleToFill" bouncesZoom="NO" style="plain" separatorStyle="default" rowHeight="44" sectionHeaderHeight="22" sectionFooterHeight="22" id="i5M-Pr-FkT">
+            <rect key="frame" x="0.0" y="0.0" width="375" height="667"/>
+            <autoresizingMask key="autoresizingMask" widthSizable="YES" heightSizable="YES"/>
+            <color key="backgroundColor" systemColor="systemBackgroundColor" cocoaTouchSystemColor="whiteColor"/>
+            <viewLayoutGuide key="safeArea" id="vLr-E1-eTs"/>
+            <connections>
+                <outlet property="dataSource" destination="-1" id="Tng-2m-Rnh"/>
+                <outlet property="delegate" destination="-1" id="9aC-8N-iBw"/>
+            </connections>
+        </tableView>
```
Comment (aswathyur) [inline IAssets/IAssets/MyAssetsTableViewController.xib:24]: Why didn't this be done in storyboard

File: IAssets/IAssets/PasswordChangeViewController.swift [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+            
+            if(passwordTextField.text == passwordConfirmationTextField.text) {
+                
+                let appDelegate = UIApplication.shared.delegate as! AppDelegate
+                let containerOfAppDelegate = appDelegate.persistentContainer
+                
+                if let userEntity = NSEntityDescription.insertNewObject(forEntityName: "User", into: containerOfAppDelegate.viewContext) as? User {
+                    
+                    userEntity.password = passwordTextField.text
+
```
Comment (aswathyur) [inline IAssets/IAssets/PasswordChangeViewController.swift:47]: This is wrong

**Pull request merged**